src/validation.h

```diff
@@ -1024,6 +1024,12 @@ class ChainstateManager
     //! ResizeCoinsCaches() as needed.
     void MaybeRebalanceCaches() EXCLUSIVE_LOCKS_REQUIRED(::cs_main);
 
+    //! @returns height at which the active UTXO snapshot was taken, if applicable.
+    std::optional<CBlockIndex*> GetSnapshotBaseBlock() EXCLUSIVE_LOCKS_REQUIRED(::cs_main);
```

Comment:
In commit "validation: fix CheckBlockIndex for multiple chainstates" (0dd5757fd7c5b1d151abe45843711780a4b3885b)

Not very important since this function is only called one place, but it seems like it would be nicer to just return `CBlockIndex*` instead of `std::optional<CBlockIndex*>`. This way callers don't need to double-dereference, and can just check whether this is null or not null, instead having to check for three states (`nullopt`, `nullptr`, or block), or having to ignore the `nullptr` case (which I assume is not currently possible)

Reply:
I was going to suggest changing to std::optional<CBlockIndex&> since the inner type is not nullable. If you want rebindable, reference_wrapper.